### PR TITLE
MGDOBR-112: add shard and webhook-robot role and allow only shards to access internal api of the manager

### PR DESCRIPTION
**Pull request by r00ta**
[MGDOBR-112](https://issues.redhat.com/browse/MGDOBR-112) 

Please make sure that your PR meets the following requirements:

- [x] Your code is properly formatted according to [this configuration](https://github.com/kiegroup/kogito-runtimes/tree/main/kogito-build/kogito-ide-config)
- [x] Your commit messages are clear and reference the JIRA issue e.g: "[MGDOBR-1] - $clear_explanation_of_what_you_did"
- [x] All new functionality is tested
- [x] Pull Request title is properly formatted: `MGDOBR-XYZ Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains link to any dependent or related Pull Request
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: infra/src/main/java/com/redhat/service/bridge/infra/api/APIConstants.java [hunk 1]
```diff
@@ -20,6 +20,11 @@
      */
     public static final String SHARD_API_BASE_PATH = "/api/v1/shard/bridges/";
 
+    /**
+     * Shard realm role.
+     */
+    public static final String SHARD_REALM_ROLE = "shard";
```
Comment (robpblake) [inline infra/src/main/java/com/redhat/service/bridge/infra/api/APIConstants.java:26]: @r00ta Can I ask you to raise this at the security workstream tomorrow. IIRC the pattern of defining roles for a Shard is going to be dropped in favour of another approach of managing which Service Accounts are Shards and should have access to the API.

**Review by robpblake — COMMENTED** [review]
@r00ta All looks good, apart from the open question of whether or not we should implement a Role for the Shard.

**Review by sutaakar — APPROVED** [review]
Looks good to me

File: manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java [hunk 2]
```diff
@@ -70,8 +85,16 @@ public Response getBridges() {
 
     @PUT
     public Response updateBridge(BridgeDTO dto) {
+        failIfNotAuthorized(identity);
         LOGGER.info("Shard wants to update the Bridge with id '{}' with the status '{}'", dto.getId(), dto.getStatus());
         bridgesService.updateBridge(dto);
         return Response.ok().build();
     }
+
+    private void failIfNotAuthorized(SecurityIdentity identity) {
```
Comment (robpblake) [inline manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java:94]: @r00ta On the next iteration of this work, perhaps we can consider a `Filter` and a CDI annotation to bind the `Filter` to this resource: https://quarkus.io/specs/jaxrs/2.1/index.html#Name_Binding

That would mean that we could annotate this Resource (or other resources) that need a valid Shard identity with something like `@ValidShard` and then we simply encapsulate the logic of checking the shard is valid in the `Filter` we implement. That would align our implementation with other conventions e.g `@Authenticated`

We don't need to do that in this PR, but might be a nice approach if we need to extend the ways in which we check a Shard is valid.

WDYT?

**Review by robpblake — APPROVED** [review]
LGTM! 🚀

File: manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java (hunk 2 above)
Comment (r00ta) [reply manager/src/main/java/com/redhat/service/bridge/manager/api/internal/ShardBridgesSyncAPI.java:94]: https://issues.redhat.com/browse/MGDOBR-245

**Review by sutaakar — APPROVED** [review]

**Pull request merged**